controller/sentry/service.go

```diff
@@ -0,0 +1,51 @@
+package sentry
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/getsentry/sentry-go"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+type Config struct {
+	Dsn    string
+	Logger micrologger.Logger
+}
+
+type Service struct {
+	enabled bool
+	logger  micrologger.Logger
+}
+
+func New(config Config) (*Service, error) {
+	disabled := Service{enabled: false}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.Dsn == "" {
+		config.Logger.Log("level", "warning", "Sentry DSN is not set.")
+		return &disabled, nil
+	}
+	config.Logger.Log("level", "debug", fmt.Sprintf("Setting up Sentry with DSN %s", config.Dsn))
+	err := sentry.Init(sentry.ClientOptions{
+		Dsn: config.Dsn,
+	})
+	if err != nil {
+		return &disabled, microerror.Mask(err)
+	}
+
+	svc := Service{
+		enabled: true,
+		logger:  config.Logger,
+	}
+
+	return &svc, nil
+}
+
+func (s *Service) Capture(ctx context.Context, err error) {
+	if s.enabled {
+		sentry.CaptureException(err)
```

Comment:
I am wondering why we need the abstraction here when we call the global library here again. Think how we use the Prometheus client. I think for the single location usage here we could do without and keep it more simple?

Reply:
There is much more that can be passed to the sentry backend (such as context about the software genereting the error event for example).
While this is the very basic usage, I expect more features to be added in the near future.
For this reason I'd prefer to have all the logic somehow "hidden" behind a simple interface.
That's the reason why I prefer having a module for this.